liquidhaskell-boot/src/Language/Haskell/Liquid/GHC/CoreToLogic.hs

```diff
@@ -30,13 +30,10 @@ coreToLogic = unlines
   , "define GHC.Classes.not x              = (~ x)"
   , "define GHC.Internal.Base.$ f x        = (f x)"
   , ""
-  , "define Language.Haskell.Liquid.Bag.get k m   = (Map_select m k)"
-  , "define Language.Haskell.Liquid.Bag.put k m   = (Map_store m k (1 + (Map_select m k)))"
-  , "define Language.Haskell.Liquid.Bag.union m n = (Map_union  m n)"
-  , "define Language.Haskell.Liquid.Bag.empty     = (Map_default 0)"
-  , ""
-  , "define Data.Map.Base.insert k v m     = (Map_store m k v)"
-  , "define Data.Map.Base.select k v       = (Map_select m k)"
+  , "define Language.Haskell.Liquid.Bag.get k b   = (Bag_count b k)"
```

Comment:
It would be better if that was a text file. Even better extensible by the user.

Reply:
It used to be a text file, but it required some contortions to find it at runtime. One of the pains was having to define an environment variable `liquidhaskell_datadir` whenever someone wanted to run liquidhaskell after building it.

If the user should be able write his owned `define` annotations, maybe they should be supported in source code, instead of constraining them to a specific file.

Reply:
@facundominguez, so if we defined the specifics for example of the `Data.Map` to a `Liquid.Haskell.Data.Map` haskell file and the user imported it together with the `Data.Map`, would that be ok? 

I think the current setting is ok for some core theories, but more specific things, like Bag could be selectively imported by the user. Esp. if we aim to define new theories. 

We could anyways merge this and open another PR if it is extracted.

Reply:
> so if we defined the specifics for example of the Data.Map to a Liquid.Haskell.Data.Map haskell file and the user imported it together with the Data.Map, would that be ok?

Perhaps the place to put them would be `Data.Map_LHAssumptions` (there is one such file for [Data.Set](https://github.com/ucsd-progsys/liquidhaskell/blob/develop/src/Data/Set_LHAssumptions.hs) already).

Looks like some changes to the code will be necessary as I don't think LH is making an effort to collect `define` annotations from other places than `CoreToLogic.hs`.